pkg/apis/pipeline/v1alpha1/build_gcs_resource.go

```diff
@@ -0,0 +1,182 @@
+/*
+Copyright 2019 The Knative Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"flag"
+	"fmt"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+var (
+	buildGCSFetcherImage = flag.String("build-gcs-fetcher-image", "gcr.io/cloud-builders/gcs-fetcher:latest",
+		"The container image containing our GCS fetcher binary.")
+	buildGCSUploaderImage = flag.String("build-gcs-uploader-image", "gcr.io/cloud-builders/gcs-uploader:latest",
+		"The container image containing our GCS uploader binary.")
+)
+
+// GCSArtifactType defines a type of GCS resource.
+type GCSArtifactType string
+
+const (
+	// GCSArchive indicates that resource should be fetched from a typical archive file.
+	GCSArchive GCSArtifactType = "archive"
+
+	// GCSManifest indicates that resource should be fetched using a
+	// manifest-based protocol which enables incremental source upload.
+	GCSManifest GCSArtifactType = "manifest"
+
+	// EmptyArtifactType indicates, no artifact type is specified.
+	EmptyArtifactType = ""
+)
+
+var validArtifactTypes = []string{string(GCSArchive), string(GCSManifest)}
+
+// BuildGCSResource describes a resource in the form of an archive,
+// or a source manifest describing files to fetch.
+// BuildGCSResource does incremental uploads for files in  directory.
+
+type BuildGCSResource struct {
+	Name           string
+	Type           PipelineResourceType
+	Location       string
+	DestinationDir string
+	ArtifactType   GCSArtifactType
+	//Secret holds a struct to indicate a field name and corresponding secret name to populate it
+	Secrets []SecretParam
+}
+
+// NewBuildGCSResource creates a new BuildGCS resource to pass to knative build
+func NewBuildGCSResource(r *PipelineResource) (*BuildGCSResource, error) {
+	if r.Spec.Type != PipelineResourceTypeStorage {
```

Comment:
Did you mean BuildGCS type here or is this type also under Storage Resource?

Reply:
its a subtype under Storage Resource.